packages/xstate-wallet/src/workflows/create-and-fund-ledger.ts

```diff
@@ -0,0 +1,184 @@
+      initial: 'initializeChannel',
+      states: {
+        initializeChannel: {
+          invoke: {
+            src: 'initializeChannel',
+            onDone: {target: 'createObjective', actions: []}
+          }
+        },
+        createObjective: {
+          invoke: {
+            src: 'createObjective',
+
+            onDone: [{target: 'done'}]
+          }
+        },
+        done: {type: 'final'}
+      },
+      onDone: 'supportPreFundState'
+    },
+
+    supportPreFundState: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getPreFundState'},
+      {src: 'supportState'},
+      'fundChannel'
+    ),
+
+    fundChannel: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getDepositingInfo'},
+      {src: 'depositing'},
+      'done'
+    ),
+
+    done: {type: 'final'},
+    failure: {type: 'final'}
+  }
+};
+
+const initializeChannel = (
+  store: Store
+): WorkflowServices['initializeChannel'] => async context => {
+  const stateVars: StateVariables = {
+    outcome: context.initialOutcome,
+    turnNum: bigNumberify(0),
+    isFinal: false,
+    appData: '0x0'
+  };
+  const entry = await store.createChannel(context.participants, CHALLENGE_DURATION, stateVars);
+  return entry.channelId;
+};
+
+const createObjective = (store: Store): WorkflowServices['createObjective'] => async context => {
+  const objective: Objective = {
+    type: 'FundLedger',
+    participants: context.participants,
+    data: {ledgerId: context.ledgerId}
+  };
+  return store.addObjective(objective);
+};
+const getPreFundState = (store: Store): WorkflowServices['getPreFundState'] => async context => {
+  const {latestState} = await store.getEntry(context.ledgerId);
+  return {state: latestState};
+};
+// TODO: Can this be shared with create-and-fund?
+const getDepositingInfo = (
+  store: Store
+): WorkflowServices['getDepositingInfo'] => async context => {
+  const {supported, myIndex} = await store.getEntry(context.ledgerId);
+  const {allocationItems} = checkThat(supported?.outcome, isSimpleEthAllocation);
+
+  const fundedAt = allocationItems.map(a => a.amount).reduce(add);
+  let depositAt = bigNumberify(0);
+  for (let i = 0; i < allocationItems.length; i++) {
+    const {amount} = allocationItems[i];
+    if (i !== myIndex) depositAt = depositAt.add(amount);
+    else {
+      const totalAfterDeposit = depositAt.add(amount);
+      return {channelId: context.ledgerId, depositAt, totalAfterDeposit, fundedAt};
+    }
+  }
+
+  throw Error(`Could not find an allocation for participant id ${myIndex}`);
+};
+
+export const options = (
+  store: Store
+): {actions: WorkflowActions; services: WorkflowServices; guards: WorkflowGuards} => {
+  return {
+    actions: {
+      assignChannelId: assign((context: WorkflowContext, event: DoneInvokeEvent<string>) => {
+        return {...context, channelId: event.data};
+      })
+    },
+    guards: {
+      doesChannelIdExist: (context: WorkflowContext) => {
+        return !!context.ledgerId;
+      }
```

Comment:
```suggestion
      doesChannelIdExist: (context: WorkflowContext) => !!context.ledgerId
```

Reply:
If we have a preference this should we enforce it with a linter? (I'm assuming there's a rule that would apply to this)

Reply:
I would prefer it in this instance only because it makes the options very clean, and IMO more readable; each line is a single option.

Reply:
In case we do prefer this (I do), we can use [this rule](https://eslint.org/docs/rules/arrow-body-style#options) with `'as-needed'`